openedx/features/course_card/helpers.py

```diff
@@ -96,3 +97,19 @@ def get_course_cards_list():
     course_card_ids = [cc.course_id for cc in cards_query_set]
     courses_list = CourseOverview.objects.select_related('image_set').filter(id__in=course_card_ids)
     return courses_list
+
+
+def check_course_part_of_specialization(course_id, programs):
+    """
+    Helper function to check if course is part of specialization
+    @param course_id: crouse key
+    @param programs: list of specialization programs
+    @return:
+    """
+    for program in programs:
+        for program_course in program['courses']:
+            if program_course['course_runs']:
+                for course_rerun in program_course['course_runs']:
+                    if course_rerun['key'] == str(course_id):
+                        return True
```

Comment:
good suggestion. Actually early exit possible at 2nd level where we are going inside course reruns but the issue is at that level the course key that we receive from api data is different than the one that we have so that don't seem to be possible. if you have any other strategy please share.

Reply:
@monofal I didn't get your point that why it isn't possible. 
Is code(with early exit) I shared above will not work for your case? 
I added early exit at this line by inverting your condition, 
```
if not program_course['course_runs']:
      continue
```